### PR TITLE
always provide revision number for xrefs

### PR DESCRIPTION
**Pull request by vladak**

### CODE REVIEW

**Comment by vladak** [comment]
I found out this does not work yet - clicking on line number erases the revision part of the URL. Needs a bit more work.

File: src/org/opensolaris/opengrok/web/PageConfig.java [hunk 1]
```diff
@@ -1190,6 +1193,50 @@ public File findDataFile() {
...
+        return rev.equals(getLatestRevision());
+    }
+
+    /**
+     * Get the location of cross reference for given file containing the current
+     * revision.
+     * @return location to redirect to or null if failed
+     */
+    public String getLatestRevisionLocation() {
+        return new String(req.getContextPath() + Prefix.XREF_P +
```
Comment (tarzanek) [inline src/org/opensolaris/opengrok/web/PageConfig.java]: use StringBuilder and append please

**Comment by tarzanek** [comment]
please fix the commend and merge

**Comment by tarzanek** [comment]
(and thank youuuuu sir! :) )

File: src/org/opensolaris/opengrok/web/PageConfig.java (hunk 1 above)
Comment (vladak) [inline src/org/opensolaris/opengrok/web/PageConfig.java]: what's the advantage of `StringBuilder` ? Seems less readable to me.

**Comment by vladak** [comment]
Still need to figure out the hash URL rewrite. I suspect there is some JS meddling with the URL.

**Pull request merged**